controllers/case-controller.js

```diff
@@ -96,9 +97,15 @@ const CaseController = (Case, sequelize) => {
         dateTo: body.dateTo || null,
         phoneNumber: body.phoneNumber ? `%${body.phoneNumber.replace(notDigits, '')}%` : null,
         counselor: body.counselor || null,
+        limit: query.limit,
+        offset: query.offset,
       },
     });
-    return { count: metadata.rowCount, cases };
+
+    const count = casesWithTotalCount.length > 0 ? casesWithTotalCount[0].totalCount : 0;
+    const cases = casesWithTotalCount.map(({ totalCount, ...rest }) => ({ ...rest }));
```

Comment:
You can add an extra condition to the ternary just in case `casesWithTotalCount` is `null` or `undefined`.

```javascript
const count = casesWithTotalCount && casesWithTotalCount.length > 0 ? casesWithTotalCount[0].totalCount : 0;
const cases = casesWithTotalCount ? casesWithTotalCount.map(({ totalCount, ...rest }) => ({ ...rest })) : [];
```